packages/starknet-snap/src/rpcs/executeTxn.ts

```diff
@@ -0,0 +1,397 @@
+    contractAddress: deployResp.contract_address,
+    contractFuncName: '',
+    contractCallData: [],
+    finalityStatus: TransactionStatus.RECEIVED,
+    executionStatus: TransactionStatus.RECEIVED,
+    status: '',
+    failureReason: '',
+    eventIds: [],
+    timestamp: Math.floor(Date.now() / 1000),
+  };
+  const transactionStateManager = new TransactionStateManager();
+  await transactionStateManager.addTransaction(txn);
+}
+
+/**
+ * The RPC handler to execute a transaction.
+ */
+export class ExecuteTxnRpc extends AccountRpcController<
+  ExecuteTxnParams,
+  ExecuteTxnResponse
+> {
+  protected requestStruct = ExecuteTxnRequestStruct;
+
+  protected responseStruct = ExecuteTxnResponseStruct;
+
+  /**
+   * Execute the transaction request handler.
+   *
+   * @param params - The parameters of the request.
+   * @param params.address - The address of the signer.
+   * @param params.calls - The invoke calls to execute
+   * @param params.abis - The abis associated to invocations.
+   * @param params.details - The detail associated to the call.
+   * @returns The InvokeFunctionResponse as an `ExecuteTxnResponse`.
+   */
+  async execute(params: ExecuteTxnParams): Promise<ExecuteTxnResponse> {
+    return super.execute(params);
+  }
+
+  protected async handleRequest(
+    params: ExecuteTxnParams,
+  ): Promise<ExecuteTxnResponse> {
+    const { address, calls, abis, details } = params;
+    const isArray = Array.isArray(calls);
+    const callsArray = isArray ? calls : [calls];
+    const invocations: Invocations = callsArray.map((call) => {
+      return {
+        type: TransactionType.INVOKE,
+        payload: call,
+      };
+    });
+
+    const estimateFeeResp = await getEstimatedFees(
+      this.network,
+      address,
+      this.account.privateKey,
+      this.account.publicKey,
+      invocations as unknown as Invocations,
+      {
+        version: details?.version ?? TRANSACTION_VERSION,
+      },
+    );
+
+    const accountDeployed = !estimateFeeResp.includeDeploy;
+
+    if (
+      !(await this.getExecuteTxnConsensus(
+        address,
+        callsArray,
+        details as UniversalDetails,
+        accountDeployed,
+      ))
+    ) {
+      throw new UserRejectedRequestError() as unknown as Error;
+    }
+
+    if (!accountDeployed) {
+      await createAccount(
+        this.network,
+        this.account.publicKey,
+        this.account.privateKey,
+        CAIRO_VERSION,
+        recordAccountDeployment,
+        true,
+      );
+    }
+
+    const transactionVersion = details?.version ?? TRANSACTION_VERSION;
+    // transaction version needs to be passed in the account
+    // having it also present in the details of the execution call creates
+    // unpredictable behaviour. This issue is not present when estimating fee bulk.
+    delete details.version;
+    const executeTxnResp = await executeTxnUtil(
+      this.network,
+      address,
+      this.account.privateKey,
```

Comment:
```suggestion
      privateKey,
```

extract from account